evalml/tests/component_tests/test_featuretools.py

```diff
@@ -0,0 +1,102 @@
+from unittest.mock import patch
+
+import featuretools as ft
+import pandas as pd
+import woodwork as ww
+
+from evalml.pipelines.components import FeatureTools
+from evalml.utils.gen_utils import _rename_column_names_to_numeric
+
+
+def test_numeric_columns(X_y_multi):
+    X, y = X_y_multi
+    X_pd = pd.DataFrame(X)
+    X_num = _rename_column_names_to_numeric(X_pd)
+
+    feature = FeatureTools()
+    feature.fit(X_num, y)
+    feature.transform(X_num)
+
+
+@patch('evalml.pipelines.components.transformers.preprocessing.featuretools.dfs')
+@patch('evalml.pipelines.components.transformers.preprocessing.featuretools.calculate_feature_matrix')
+def test_index(mock_dfs, mock_calculate_feature_matrix, X_y_multi):
+    X, y = X_y_multi
+    X_pd = pd.DataFrame(X)
+    X_new_index = X_pd.copy()
+    index = [i for i in range(len(X))]
+    new_index = [i * 2 for i in index]
+    X_new_index['index'] = new_index
+
+    # check if _make_entity_set keeps the intended index
+    feature = FeatureTools()
+    feature.fit(X_new_index)
+    feature.transform(X_new_index)
+    arg_es = mock_dfs.call_args[1]['entityset'].entities[0].df['index']
+    arg_tr = mock_calculate_feature_matrix.call_args[1]['entityset'].entities[0].df['index']
+    assert arg_es.to_list() == new_index
+    assert arg_tr.to_list() == new_index
+
+    # check if _make_entity_set fills in the proper index values
+    feature.fit(X_pd)
+    feature.transform(X_pd)
+    arg_es = mock_dfs.call_args[1]['entityset'].entities[0].df['index']
+    arg_tr = mock_calculate_feature_matrix.call_args[1]['entityset'].entities[0].df['index']
+    assert arg_es.to_list() == index
+    assert arg_tr.to_list() == index
+
+
+def test_input_types(X_y_binary, X_y_multi, X_y_regression):
+    datasets = locals()
+    for dataset in datasets.values():
+        X, y = dataset
+
+        feature = FeatureTools()
+        feature.fit(X, y)
+        feature.transform(X)
+
+        X_pd = pd.DataFrame(X)
+        feature.fit(X_pd, y)
+        feature.transform(X_pd)
+
+        X_ww = ww.DataTable(X_pd)
+        feature.fit(X_ww)
+        feature.transform(X_ww)
+
+
+def test_transform(X_y_binary, X_y_multi, X_y_regression):
```

Comment:
I think it's ok to keep the tests as-is but I think we can condense this one and `test_input_types` with `@parametrize`!